### PR TITLE
Remove protractor E2E tests

### PR DESCRIPTION
**Pull request by oliverguenther**
In agreement with @furinvader, we have decided to remove protractor
testing from OpenProject.

Frontend tests are implemented in karma and test single units of work as before.
Since the freshness of APIv3 JSON mocks are getting ouf hand and we have covered almost all E2E protractor tests with Rails, we're going to resort to Rails to write concise _actual_ integration test with correct data.

The only argument _for_ protractor I see is that users proficient in Javascript only can write E2E tests. I don't think that makes a lot of sense given the complexity of our application, and learning to write integration tests in Ruby isn't that hard.

The downsides, on the other hand, are manifold:
1. Protractor requires us to maintain a set of mock responses from the APIv3. If we are changing only a few lines of backend code, we'll have to generate these again or risk false negatives, since the API responses are not identical to the mocks.
2. Protractor is damn slow for the few test cases we're running. I argue that the full Rails stack is performing in the same order of magnitude, and that's with the full stack and actual data.
3. A green protractor suite does not actually provide re-assurance on an integration level. Due to the mocks, it's just large abstraction to unit testing. We can do the same thing faster with karma.

Following this PR, I'm going to replace the few deleted mock-based tests with backend integration tests.

RFC @opf/developers

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://oliverguenther-housekeeping-remove-protractor.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-housekeeping-remove-protractor.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

**Comment by furinvader** [comment]
THANK YOU
I'm glad we can get rid of that redundant testing framework.

File: frontend/.jshintrc [hunk 1]
```diff
@@ -10,7 +10,6 @@
     "angular",
     "inject",
     "module",
-    "protractor",
```
Comment (furinvader) [inline frontend/.jshintrc:13]: We probably can get rid of `expressjs` too

Comment (oliverguenther) [inline frontend/.jshintrc:13]: express runs the living styleguide. You can export it with `gulp styleguide`, but that won't load all assets.

**Pull request merged**